src/hardware/adc.rs

```diff
@@ -123,38 +115,41 @@ impl Adc {
         // SPI at 1 MHz. SPI MODE_0: idle low, capture on first transition
         let spi: Spi<_, _, u8> = spi4.spi(pins.spi, spi::MODE_0, 12500.khz(), spi4_rec, clocks);
```

Comment:
It's not 1 MHz (don't write redundant comments repeating the code).
It's not idle low and not capturing on first transition. Hence probably not MODE 0. How did you choose this?

![image](https://user-images.githubusercontent.com/1338946/162440307-7ed6b516-0380-42ab-813e-cfde897f46c5.png)

![image](https://user-images.githubusercontent.com/1338946/162440911-9a792492-f5f6-452a-9154-1885bae2f733.png)

Reply:
Yeah... Sorry I forgot how I came to this result. We had some discussion about which modes are possible and I just went ahead with one without crosschecking again with the datasheet. And "idle low, capture first" accedentally lead to valid data. Unfortunately "idle high, capture second" now leads to invalid data. I'll look into it now.

Reply:
nevermind, I just used the wrong mode still.